File: final-project-movie-portal/backend/src/main/java/com/furkanbilgin/finalproject/movieportal/service/CommentService.java
package com.furkanbilgin.finalproject.movieportal.service;

import com.furkanbilgin.finalproject.movieportal.dto.social.CommentDTO;
import com.furkanbilgin.finalproject.movieportal.dto.user.comment.UserCommentDTO;
import java.util.List;

public interface CommentService {
  List<CommentDTO> getCommentsByMovie(Long movieId);

  CommentDTO addComment(String username, Long movieId, UserCommentDTO userCommentDTO);

  void deleteComment(Long commentId);

  boolean isCommentOwner(Long commentId, String username);
}

File: final-project-movie-portal/frontend/src/api/services/commentService.ts
import { api } from '..';
import {
  addComment,
  deleteComment,
  getCommentsByMovie,
  type CommentDto,
} from '../../client';

export interface CommentService {
  getByMovie: (id: number) => Promise<CommentDto[]>;
  create: (
    id: number,
    comment: string,
    parentCommentId?: number
  ) => Promise<void>;
  delete: (id: number) => Promise<void>;
}

export const commentService = {
  getByMovie: async (id: number) => {
    const res = await getCommentsByMovie({
      path: { movieId: id },
      client: api.client,
    });
    return res.data ?? [];
  },
  create: async (id: number, comment: string, parentCommentId?: number) => {
    await addComment({
      body: { content: comment, parentCommentId },
      path: { movieId: id },
      client: api.client,
    });
  },
  delete: async (id: number) => {
    await deleteComment({
      body: { commentId: id },
      client: api.client,
    });
  },
} as CommentService;
